src/frontends/pytorch/src/transforms/aten_index_replacer.cpp

```diff
@@ -108,10 +108,10 @@ AtenIndexToSelect::AtenIndexToSelect() {
                         continue;
                     }
                 }
-                auto id_dtype = ids[i].get_node_shared_ptr()->get_element_type();
+                auto id_dtype = ids[i].get_element_type();
                 if (id_dtype == element::boolean || id_dtype == element::u8) {
```

Comment:
Does this work as expected now? Or is it required for types to be know during conversion (so we have to set them in InputModel)?

Reply:
I'm not sure that we can work without type knowledge at all, but I assume in all cases where bool logic should be, it will be already known (parent op returns bool - e.g. where, or some logical ops eq, le, get e.t.c.)